R/object_name_linters.R

```diff
@@ -29,14 +29,19 @@ object_name_linter <- function(styles = c("snake_case", "symbols")) {
     xml <- source_file$full_xml_parsed_content
 
     xpath <- paste0(
-      # Left hand assignments
-      "//expr[SYMBOL or STR_CONST][following-sibling::LEFT_ASSIGN or following-sibling::EQ_ASSIGN]/*",
-
-      # Or
-      " | ",
-
-      # Right hand assignments
-      "//expr[SYMBOL or STR_CONST][preceding-sibling::RIGHT_ASSIGN]/*",
+      # assignments
+      "//SYMBOL[",
+      " not(preceding-sibling::OP-DOLLAR)",
+      " and ancestor::expr[",
+      "  following-sibling::LEFT_ASSIGN",
+      "  or preceding-sibling::RIGHT_ASSIGN",
+      "  or following-sibling::EQ_ASSIGN",
+      " ]",
+      " and not(ancestor::expr[",
```

Comment:
as it's written, this is a totally different expression -- is there some relationship between the `LEFT_ASSIGN` (etc) ancestor and the `OP-LEFT-BRACKET` ancestor? I'm wondering if it's safer to express both conditions at once (it might not be).

Anyway I am running on a sample of 500 packages now, let's see what comes out.

Reply:
The problem with combining it the conditions need to be nested, i.e. there may be no such ancestor before reaching the (first) assignment ancestor.

I couldn't get it to work in a single condition but if you habe an idea for a different one, I'm happy to try it.

The tests are pretty comprehensive now. The only missing bits are nested expressions with on-the-fly assignments, but I checked a few of these interactively, e.g. `fun(BAD <- 42)`.